src/lib_json/json_value.cpp

```diff
@@ -182,6 +182,15 @@ Value::CZString::CZString(const CZString& other)
                      ? noDuplication : duplicate)
                  : other.index_) {}
 
+#ifdef JSON_HAS_RVALUE_REFERENCES
+Value::CZString::CZString(CZString&& other)
+	: cstr_(other.cstr_)
+	, index_(other.index_)
```

Comment:
@lanzkron, personally I like your `,` under `:`, since it makes adding arguments easier, but we try to stick with google-style-guide to avoid arguments. (Actually, it's best if you reformat with Clang, since @jacobsa, did a lot of work on that.) In general, I trust @BillyDonahue's judgement, so I thank you for taking all his suggestions.

I want to pull this and fix the the merge issues myself, and maybe reformat with Clang. However, I don't like adding pre-processor macros. I wonder if we wouldn't be better off with a `2.0.0-alpha` branch. We can guarantee *source-code* compatibility but completely relax *binary* compatibility. We can require compilers which support all features. We could then add all sorts of things before it is stable. Anybody can use `1.y.z` or even `0.y.z` if they have a problem.

Downside: Chromium would not have move-constructors. But I don't think Chromium cares about binary-compatibility, so maybe they could simply upgrade.

Opinions?